docs/stdlib/json.rst

```diff
@@ -219,15 +216,13 @@ possible to cast a JSON value directly into a :eql:type:`tuple`.
 
 .. eql:operator:: jsonplus: json ++ json -> json
 
-    JSON concatenation.
-
-    JSON arrays, objects and strings can be concatenated with JSON values of
-    the same type into a new JSON value.
-
-    If you concatenate two JSON objects, you get a new object whose keys will
-    be a union of the keys of the input objects. If a key is present in both
-    objects, the value from the second object is taken.
+    Concatenates given types of :eql:type:`json` and :eql:type:`Object`
+    into one.
 
+    This results in a reference of both array's elements conjoined together.
+    If there are mutual keys in the given :eql:type:`json`, the value of the
+    second one will be used instead:
```

Comment:
```suggestion
    The result is a new :eql:type:`json` value containing the elements of both
    concatenated values. If concatenated objects have identical keys, the value
    of the second one will be used.
    
    Examples:
```

- Result is not a reference (at least, not from the user's perspective) but a new value.
- Clarified some of the language
- Clarified that key overrides apply only to `json` objects
- Added an "Examples" since the examples don't demonstrate only the idea conveyed in the last sentence. Dropping a colon on that suggests they do.